### ISSUE TITLE
Nightmare freezes on either evaluate() or screenshot()

### ISSUE THREAD

**Issue by ryanvincent29** [segmentio/nightmare#736]
I am working on moving from Phantom JS to Nightmare in an effort to get more CSS compatibility on my pages.

As a high-level view, my Nightmare instance lives inside a Docker container and listens for data to be pushed onto a queue. Once that data is pushed, I pull some data (including a string of HTML) out of the queue, alter the page a bit using Cheerio, and then turn the data into a HTML Data URL. 

Then, I goto the data url, use a couple of evaluate functions, scroll the page, and take a screenshot.

Here is a condensed version of my code:

``` javascript
function callNightmare(nightmare, html, data) {
  return new Promise(function(resolve, reject){
    var imageLoc = "./screenshot" + data.randomGuid + '.png';
    nightmare
      .userAgent(data.agentString)
      .goto(html)
      .wait()
      .then(function() {
        //remove scrollbar from image
        return nightmare.evaluate(function() { 
          var s = document.styleSheets[0];
          s.insertRule('::-webkit-scrollbar { display:none; }'); }
      })
      .then(function() {
        return nightmare.viewport(data.width, data.height);
      })
      .then(function() {
        return nightmare.inject('js', './localScript.js');
      })
      .then(function() {
        return nightmare.evaluate(getScrollPositionFromPage);
      })
      .then(function(scrollPosition) {
        return nightmare.scrollTo(scrollPosition.top, scrollPosition.left);
      })
      .then(function() {
        return nightmare.evaluate(fillInputsOnPageWithValues);
      })
      .then(function(){
        return nightmare.screenShot(imageLoc);
      })
      .then(function() {
        return nightmare.end();
      })
      .then(function() {
        resolve(imageLoc);
      })
      .catch(function(err) { 
        reject(err); 
      })
  });
}
```

However, my script will occasionally lock up during various evaluate functions, or during the screenshot function. Other times, the script runs successfully. I have tried this on various websites with the same results.

I can't pinpoint anything in the script that is causing this issue, and I didn't know if any of you had any thoughts on the problem?

**Comment by rosshinkley** [comment]
Silly question: have you tried running with `DEBUG=*` to see if that has any helpful information?

**Comment by rosshinkley** [comment]
(Also: Possibly related to #405.)

**Comment by ryanvincent29** [comment]
Yeah, I run it with DEBUG=*, it outputs 2 lines indicating the initialization of Nightmare, but then I stop getting output from that debug path for the remainder of the program.

**Comment by rosshinkley** [comment]
Considering this fixed with #927.  If you're still having this problem with that change, feel free to reopen.
